### PR TITLE
docs(routing): improvements

### PR DESCRIPTION
**Pull request by Atinux**
### ❓ Type of change

- [x] 📖 Documentation (updates to the documentation or readme)

### 📚 Description

- Renamed `Routing` in **Usage** to `NuxtLink`
- Improved the content of the `pages/` directory in **Directory Structure**

### CODE REVIEW

**Review by danielroe — APPROVED** [review]
This is a really nice improvement.

File: docs/content/3.docs/1.usage/9.nuxt-link.md [hunk 1]
```diff
@@ -1,10 +1,8 @@
-# Routing
-
-## `<NuxtLink>`
+# NuxtLink
```
Comment (danielroe) [inline docs/content/3.docs/1.usage/9.nuxt-link.md:1]: I think the idea was that we might cover more on this page than just `<NuxtLink>`

Comment (Atinux) [reply docs/content/3.docs/1.usage/9.nuxt-link.md:1]: I think we shall keep it simple since it will be moved to API Reference later on :)

Comment (Atinux) [reply docs/content/3.docs/1.usage/9.nuxt-link.md:1]: And the `pages/` file should become the Routing guide.

**Review by pi0 — APPROVED** [review]

**Pull request merged**